app/assets/javascripts/app/views/list_view.coffee

```diff
@@ -0,0 +1,62 @@
+class App.views.ListView extends Backbone.View
+  className: 'list'
+  tagName: 'ol'
+
+  initialize: (options) ->
+    @items = options.items
+    @parent = options.parent
+    @$el.data('parent-id', @parent)
+
+  handleToggleButton: (
+    parentId, originalParentId, siblings, endingPosition) ->
+
+    # Ensure that the new parent's toggle-children button displays.
+    $('#item-' + parentId).find('.toggle-children').first().show()
+
+    # If the original parent doesn't have children anymore, hide its
+    # toggle-children button.
+    if $('#item-' + originalParentId).find('ol li').length == 0
+      $('#item-' + originalParentId).find('.toggle-children').first().hide()
+
+  # Sets the new positions and parent. It's important to go through all
+  # sibling elements to ensure their positions never overlap in case of
+  # multiple insertions into the same list index.
+  handleNewPositions: (siblings, endingPosition, parentId, movedId) ->
+    siblings.each (idx, item) ->
+      itemID = $(item).data('item-id')
+      current = new App.models.Item({id: itemID})
+      if idx < endingPosition
+        current.set({position: idx})
+      else if idx > endingPosition
+        current.set({position: idx + 1})
+      idx += 1
+
+    model = new App.models.Item({id: movedId})
+    model.set({
+      position: endingPosition,
+      parent_id: parentId
+    })
+
+  addSortable: ->
+    that = this
```

Comment:
in coffee, aliasing `this` to a temp value is usually no needed, all you need to do is use fat arrow `=>` when defining a function